### PR TITLE
import patches for nRF51822 by Christopher Mason

### PR DESCRIPTION
**Pull request by floe**
As mentioned on the other thread, these are simply the patches from @cmason1978 imported into a new fork of the programmer. My git-fu wasn't strong enough to figure out how to apply these across the repo split :-)

### CODE REVIEW

**Comment by cmason1978** [comment]
Thanks for doing this! My mchck work is a bit on hold right now.

**Comment by floe** [comment]
My pleasure, thanks for writing them. As mentioned elsewhere, the patch is also very useful for a ultra-minimal nRF51 dev environment with just a breakout board and a Buspirate. When I get around to it, I'll also add support for the UICR block.

File: nrf51.rb [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    unsigned :PPFC, 0x02C, :desc => "Pre-programmed factory code present"
+
+    unsigned :NUMRAMBLOCK, 0x034, :desc => "Number of individually controllable RAM blocks"
+    unsigned :SIZERAMBLOCK, 0x038, :vector => 4, :desc => "Size of RAM block n in bytes"
+
+  end
+
+  #0x10001000 UICR UICR User Information Configuration Registers
+
+  def mass_erase
```
Comment (corecode) [inline nrf51.rb]: should this call @nvmc.mass_erase?

Comment (floe) [inline nrf51.rb]: Good catch, fixed.

**Comment by corecode** [comment]
Just to make sure, because I don't have a programmer<>nfr51 setup right now:  did you test this code?  Did you manage to test the changes with the k20 chips?

**Comment by floe** [comment]
Yes, I've successfully flashed an nRF51 with this code using a Buspirate (both softdev & regular code). Mass erase also works now. I don't have a mchck myself, so I couldn't test it on k20...
-- 
SENT FROM MY PDP-11

**Pull request merged**